test/Concurrency/issue-80363.swift

```diff
@@ -0,0 +1,47 @@
+// RUN: %target-typecheck-verify-swift
+
+// https://github.com/swiftlang/swift/issues/80363
+
+class C {}
+
+func testLocal() {
+  isolated let c = C()
+  // expected-error@-1 {{'isolated' may only be used on 'deinit' declarations}}
```

Comment:
we should fix the message in a follow up; `isolated` is also legal on isolated function parameters but this change is lgtm, at least we diagnose the mis-use now.